fe/fe-core/src/main/java/org/apache/doris/analysis/ShowAnalyzeStmt.java

```diff
@@ -0,0 +1,340 @@
+        return -1L;
+    }
+
+    public long getOffset() {
+        if (this.limitElement != null && this.limitElement.hasOffset()) {
+            return this.limitElement.getOffset();
+        }
+        return -1L;
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
+        super.analyze(analyzer);
+
+        String dbName = null;
+        String tblName = null;
+        if (this.dbTableName != null) {
+            dbName = this.dbTableName.getDb();
+            tblName = this.dbTableName.getTbl();
+        }
+
+        if (Strings.isNullOrEmpty(dbName)) {
+            dbName = analyzer.getDefaultDb();
+        } else {
+            dbName = ClusterNamespace.getFullName(analyzer.getClusterName(), dbName);
+        }
+
+        if (Strings.isNullOrEmpty(dbName)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+        }
+        this.db = analyzer.getCatalog().getDbOrAnalysisException(dbName);
+
+        if (Strings.isNullOrEmpty(tblName)) {
+            this.table = null;
+            List<Table> tables = this.db.getTables();
+            for (Table table : tables) {
+                checkShowAnalyzePriv(dbName, table.getName());
+            }
+        } else {
+            this.table = this.db.getTableOrAnalysisException(tblName);
+            checkShowAnalyzePriv(dbName, this.table.getName());
+        }
+
+        // analyze where clause if not null
+        if (this.whereClause != null) {
+            if (this.whereClause instanceof CompoundPredicate) {
+                CompoundPredicate cp = (CompoundPredicate) this.whereClause;
+                if (cp.getOp() != CompoundPredicate.Operator.AND) {
+                    throw new AnalysisException("Only allow compound predicate with operator AND");
+                }
+                // check whether left.columnName equals to right.columnName
+                checkPredicateName(cp.getChild(0), cp.getChild(1));
+                analyzeSubPredicate(cp.getChild(0));
+                analyzeSubPredicate(cp.getChild(1));
+            } else {
+                analyzeSubPredicate(this.whereClause);
+            }
+        }
+
+        // analyze order by
+        if (this.orderByElements != null && !this.orderByElements.isEmpty()) {
+            this.orderByPairs = new ArrayList<>();
+            for (OrderByElement orderByElement : this.orderByElements) {
+                if (!(orderByElement.getExpr() instanceof SlotRef)) {
+                    throw new AnalysisException("Should order by column");
+                }
+                SlotRef slotRef = (SlotRef) orderByElement.getExpr();
+                int index = analyzeColumn(slotRef.getColumnName());
+                OrderByPair orderByPair = new OrderByPair(index, !orderByElement.getIsAsc());
+                this.orderByPairs.add(orderByPair);
+            }
+        }
+    }
+
+    @Override
+    public ShowResultSetMetaData getMetaData() {
```

Comment:
since the `ShowResultSetMetaData` is always same, we could use static variable to avoid to generate `ShowResultSetMetaData` every time.

Reply:
got it